### ISSUE TITLE
feature_request(html): doesn't work MD026, if inline HTML

### ISSUE THREAD

**Issue by Kristinita** [markdownlint/markdownlint#196]
### 1. Summary

In header.

### 2. Environment

+ Windows 10 Enterprise LTSB 64-bit EN
+ Ruby 2.5.1.1
+ mdl 0.4.0

### 3. Configuration

Example project.

`.mdlrc`:

```ruby
style "mdl.rb"
```

`mdl.rb`:

```ruby
all
rule 'MD026', :punctuation => ".,;:!"
exclude_rule 'MD033'
exclude_rule 'MD041'
```

`SashaInlineHTML.md`:

```markdown
<a id="Sasha-exclamation-mark"></a>
# Sasha exclamation mark!

<a id="Sasha-question-mark"></a>
---

## Sasha question mark?
```

`SashaNoInlineHTML.md`:

```markdown
# Sasha exclamation mark!

---

## Sasha question mark?
```

### 4. Steps to reproduce

    mdl SashaNoInlineHTML.md

    mdl SashaInlineHTML.md

### 5. Expected behavior

```shell
D:\SashaDemoRepositories\SashaMdlDebugging>mdl SashaNoInlineHTML.md
SashaNoInlineHTML.md:1: MD026 Trailing punctuation in header
```

### 6. Actual behavior

```shell
D:\SashaDemoRepositories\SashaMdlDebugging>mdl SashaInlineHTML.md
```

Markdownlint no detect errors/warnings.

Thanks.
